### ISSUE TITLE
Unrecognized Locale

### ISSUE THREAD

**Issue by huujiee** [yasirkula/UnityMobileLocalizedAppTitle#3]
After uploading the app to the appstore, I received a warning:

> Unrecognized Locale - The locale names used in localization directories at ( "Payload/Project.app/Frameworks/UnityFramework.framework/zh-rCN.lproj", "Payload/Project.app/Frameworks/UnityFramework.framework/zh-rTW.lproj", "Payload/Project.app/zh-rCN.lproj", "Payload/Project.app/zh-rTW.lproj" ) are invalid. iTunes supports BCP47 but not the UN M.49 specification. Refer to the Language and Locale Designations guide at https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html for more information on naming your language-specific directories.

It seems that iOS does not support LanguageCode "zh-rCN, zh-rTW".

After my attempts, I found that "zh-Hans, zh-Hant" is available on iOS, but an error is reported when compiling on the Android platform. "zh-rCN, zh-rTW" will have the above warning after uploading to the AppStore on the iOS platform, but it will be normal on the Android platform.

I'm not sure if I'm using it incorrectly, how to solve this problem correctly, looking forward to your reply, thank you.

Here is my "LocalizedAppTitle.json":

```json
{
    "LocalizeAppNameOnAndroid": true,
    "LocalizeAppIconOnAndroid": false,
    "LocalizeAppNameOniOS": true,
    "LocalizeAppIconOniOS": false,
    "LocalizedData": [
        {
            "LanguageCode": "en",
            "AppName": "English",
            "AppShortName": "English",
            "AndroidIcon": {
                "instanceID": 0
            },
            "iOSIcon": {
                "instanceID": 0
            }
        },
        {
            "LanguageCode": "zh-rCN",
            "AppName": "简体中文",
            "AppShortName": "简体中文",
            "AndroidIcon": {
                "instanceID": 0
            },
            "iOSIcon": {
                "instanceID": 0
            }
        },
        {
            "LanguageCode": "zh-rTW",
            "AppName": "繁體中文",
            "AppShortName": "繁體中文",
            "AndroidIcon": {
                "instanceID": 0
            },
            "iOSIcon": {
                "instanceID": 0
            }
        }
    ],
    "DefaultLocalizedData": 0,
    "AndroidNameResource": "app_name",
    "AndroidIconResource": "app_icon"
}
```

**Comment by yasirkula** [comment]
According to [this](https://github.com/jcraane/kmm-resources/issues/8), zh-CN and zh-TW may work (just removing 'r'). Could you give it a try? If that doesn't work, I guess I'll have to hardcode zh-Hans and zh-Hant.

**Comment by huujiee** [comment]
Thanks, I'll try it next time I submit an updated version of the app.

**Comment by huujiee** [comment]
Sorry for not replying for so long. I tested it on the local device before and it worked normally. After I submitted the app to the AppStore for review, I no longer received the "Unrecognized Locale" warning. So far, AppStore review specialists have not approved my app, prompting that more time is needed for review, so I have not fully confirmed whether this problem has been resolved. But I'm sure it can work locally. Thank you very much for your help!
